test/SourceKit/Indexing/index_effective_access_level.swift

```diff
@@ -0,0 +1,37 @@
+// RUN: %sourcekitd-test -req=index %s -- -Xfrontend -serialize-diagnostics-path -Xfrontend %t.dia %s | %sed_clean > %t.response
+// RUN: %diff -u %s.response %t.response
+
+public enum PublicEn {
+    case a
+    case b
+}
+
+enum InternalEn {
+    case a
+    case b
+}
+
+private enum PrivateEn {
+    case a
+    case b
+}
+
+public extension PublicEn {
+    func puFoo() {}
+}
+
+extension PublicEn {
+    public func puFoo() {}
+}
+
+extension InternalEn {
+    func foo() {}
+}
+
+private extension PrivateEn {
+    func prFoo() {}
+}
+
+extension PrivateEn {
+    private func prFoo() {}
+}
```

Comment:
This file is missing salient test cases. Please include the scenarios I've enumerated earlier, where members have higher declared access levels than their containing type, and where members have `private` access levels that give them lower visibility than `fileprivate` which cannot be expressed.